eiffel-syntax-and-usage/glossary.md

```diff
@@ -0,0 +1,102 @@
+   limitations under the License.
+--->
+
+# Eiffel Glossary
+This a non-exhaustive, alphabetical list of terms used in the [Eiffel vocabulary](../eiffel-vocabulary/) and its documentation.
+
+ - [Terms Used in Event Definitions](#terms-used-in-event-definitions)
+   - [Activity](#activity)
+   - [Artifact](#artifact)
+   - [Composition](#composition)
+   - [Confidence Level](#confidence-level)
+   - [Environment](#environment)
+   - [Event](#event)
+   - [Source Change](#source-change)
+   - [Submit](#submit)
+
+ - [Terms Used in Documentation](#terms-used-in-documentation)
+   - [Pipeline](#pipeline)
+
+## Terms Used in Event Definitions
+These terms are either part of the event names in the Eiffel Vocabulary, or they are part of parameter names or values in the events. Users of Eiffel should be able to rely on these terms to be kept, unless a major change to event types are done.
+
+### Activity
+An _activity_ is any kind of action in a CI/CD system, normally triggered by an operation done in an SCM system or by a previous activity. The Eiffel protocol declares events related to activities to describe what is going on, what has finished, when did it happen and with what outcome.
+
+Activities are hierarchical. Activities could be whole [pipelines](#pipeline), or sequential or parallel parts thereof. Such pipeline parts could for example be build activities or test activities.
+
+An activity could also be an SCM operation such as a manual code review or automated source change check.
+
+Some activity types have their own specific events, such as EiffelTestCase\*Events notifying test case executions, while others are more generic and can be notified using EiffelActivity\*Events.
+
+#### Examples of events related to activities:
+- [EiffelActivityTriggeredEvent](../eiffel-vocabulary/EiffelActivityTriggeredEvent.md)
+- [EiffelActivityStartedEvent](../eiffel-vocabulary/EiffelActivityStartedEvent.md)
+- [EiffelActivityFinishedEvent](../eiffel-vocabulary/EiffelActivityFinishedEvent.md)
+- [EiffelTestCaseTriggeredEvent](../eiffel-vocabulary/EiffelTestCaseTriggeredEvent.md)
+- [EiffelTestCaseStartedEvent](../eiffel-vocabulary/EiffelTestCaseStartedEvent.md)
+- [EiffelTestCaseFinishedEvent](../eiffel-vocabulary/EiffelTestCaseFinishedEvent.md)
+
+### Artifact
+_Artifacts_ are items or software packages generated in a CI/CD [pipeline](#pipeline), for example a built binary or a Docker image. An artifact should be possible to identify using a purl (package URL). An artifact is often the subject of a test executed or a delivery performed within a CI/CD [pipeline](#pipeline).
+
+#### Examples of events related to artifacts:
+- [EiffelArtifactCreatedEvent](../eiffel-vocabulary/EiffelArtifactCreatedEvent.md)
+
+### Composition
+A _composition_ is a grouping of [artifacts](#artifact) and/or [source changes](#source-change). A composition is more or less the same as what is elsewhere sometimes referred to as a _baseline_.
```

Comment:
What is the definition of a baseline?

Reply:
With a quick googling of "baseline definition" I got a lot of similar results like this: "A baseline is a fixed reference point used for comparison." In my opinion that is enough, or do you think we should explicitly define it here also?

Reply:
I do not intend to define what a baseline is. That's why I wrote that it is elsewhere referred to as a baseline. I thought that could add to the understanding of what we mean with composition. I'll elaborate it a bit more in the definition.
Do you think I should not use _italic_ when mentioning a baseline? I wanted it to stand out as some kind of a reserved word even though it is not a term explicitly defined here.

Reply:
With a baseline, I guess you mean a definition of products and their versions. E.g. A v1, B v2, Cv1.  Maybe give an example to be more clear.

Reply:
@m-linner-ericsson , have you seen the update on this definition I made yesterday? It is hopefully more clear now. Please review that one instead.

Reply:
Yes that is better